test/functional/tests/stats/test_ioclass_stats.py

```diff
@@ -127,44 +171,158 @@ def test_ioclass_stats_sum():
+
+    with TestRun.step("Load random IO class configuration for each cache"):
+        for cache in caches:
+            random_list = IoClass.generate_random_ioclass_list(ioclass_config.MAX_IO_CLASS_ID + 1)
+            IoClass.save_list_to_config_file(random_list, add_default_rule=False)
+            cache.load_io_class(ioclass_config.default_config_file_path)
+
+    with TestRun.step(f"Validate displayed {stat_filter.name} statistics for every configured IO "
+                      f"class for all {'cores' if per_core else 'caches'}"):
+        for cache in caches:
+            with TestRun.group(f"Cache {cache.cache_id}"):
+                for core in cache.get_core_devices():
+                    core_info = f"Core {core.cache_id}-{core.core_id} ," if per_core else ""
+                    for class_id in range(ioclass_config.MAX_IO_CLASS_ID + 1):
+                        with TestRun.group(core_info + f"IO class id {class_id}"):
+                            statistics = (
+                                core.get_statistics_flat(class_id, [stat_filter]) if per_core
+                                else cache.get_statistics_flat(class_id, [stat_filter]))
+                            validate_statistics(statistics, stat_filter, per_core)
+                            if stat_filter == StatsFilter.conf:  # no percentage statistics for conf
+                                continue
+                            statistics_percents = (
+                                core.get_statistics_flat(
+                                    class_id, [stat_filter], percentage_val=True) if per_core
+                                else cache.get_statistics_flat(
+                                    class_id, [stat_filter], percentage_val=True))
+                            validate_statistics(statistics_percents, stat_filter, per_core)
+                    if not per_core:
+                        break
+
 
+def get_checked_statistics(stat_filter: StatsFilter, per_core: bool):
+    if stat_filter == StatsFilter.conf:
+        return config_stats_ioclass
+    if stat_filter == StatsFilter.usage:
+        return usage_stats
+    if stat_filter == StatsFilter.blk:
+        return block_stats_core if per_core else block_stats_cache
+    if stat_filter == StatsFilter.req:
+        return request_stats
 
-def prepare():
+
+def validate_statistics(statistics: dict, stat_filter: StatsFilter, per_core: bool):
+    for stat_name in get_checked_statistics(stat_filter, per_core):
+        if stat_name not in statistics.keys():
+            TestRun.LOGGER.error(f"Value for {stat_name} not displayed in output")
+        else:
+            del statistics[stat_name]
+    if len(statistics.keys()):
+        TestRun.LOGGER.error(f"Additional statistics found: {', '.join(statistics.keys())}")
+
+
+def prepare(cache_count=1, cores_per_cache=1):
+    cache_modes = [CacheMode.WT, CacheMode.WB, CacheMode.WA, CacheMode.WO]
+    random_cls = random.choice(list(CacheLineSize))
```

Comment:
Please pass randomized parameter via `parametrize()` to preserve test reproducibility.